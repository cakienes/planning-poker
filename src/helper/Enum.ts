export enum UserStoryStatusEnum {
    VOTED = 'VOTED',
    ACTIVE = 'ACTIVE',
    NOT_VOTED = 'NOT_VOTED',
}
