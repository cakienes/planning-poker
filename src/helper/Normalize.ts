const maxLength: any = (max: number) => (value: string) => (value && value.length > max ? value.slice(0, max) : value);

export { maxLength };
