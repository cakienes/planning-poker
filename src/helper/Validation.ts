const required: any = (value: any) => (value ? undefined : 'Please fill');

const onlyNumber: any = (value: any) => {
    return value && value !== '0' && parseInt(value, 10) ? undefined : 'Must be a number';
};

export { required, onlyNumber };
