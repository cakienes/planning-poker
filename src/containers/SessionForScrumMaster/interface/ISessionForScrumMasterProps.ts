import { RouteComponentProps } from 'react-router';
import { UserTypeEnum } from '../../../helper/Enum';
import ISession from '../../../interfaces/ISession';
import IUserStory from '../../../interfaces/IUserStory';

export default interface ISessionForScrumMasterProps extends RouteComponentProps<any> {
    activeUserStory?: IUserStory;
    getSessionsFromLocalStorage: () => void;
    giveStoryPoint: (userType: UserTypeEnum, storyPoint?: number) => void;
    selectedSession?: ISession;
    setSelectedSession: (id: string) => void;
}
