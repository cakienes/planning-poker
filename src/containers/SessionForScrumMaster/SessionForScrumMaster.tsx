import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import ActiveStory from '../../components/ActiveStory/ActiveStory';
import UserStoryList from '../../components/UserStoryList/UserStoryList';
import { UserTypeEnum } from '../../helper/Enum';
import IGlobalState from '../../interfaces/IGlobalState';
import { getSessionsFromLocalStorage, giveStoryPoint, setSelectedSession } from '../../redux/session/session.actions';
import { getActiveUserStory, getSelectedSession } from '../../redux/session/session.selectors';
import Layout from '../Layout/Layout';
import ISessionForScrumMasterProps from './interface/ISessionForScrumMasterProps';
import ScrumMasterPanel from './ScrumMasterPanel/ScrumMasterPanel';

export class SessionForScrumMaster extends React.Component<ISessionForScrumMasterProps, any> {
    componentDidMount() {
        const {
            match: {
                params: { id },
            },
        } = this.props;
        if (id) {
            this.props.setSelectedSession(id);
        }
        setInterval(this.reRenderComponent, 1000);
    }

    render() {
        const { selectedSession, activeUserStory } = this.props;
        console.log('rendered');
        return (
            <Layout>
                <div className="session">
                    <div className="row">
                        <div className="col">
                            <UserStoryList selectedSession={selectedSession} />
                        </div>
                        <div className="col-4">
                            <ActiveStory
                                selectedSession={selectedSession}
                                giveStoryPoint={this.props.giveStoryPoint}
                                activeUserStory={activeUserStory}
                                type={UserTypeEnum.SCRUM_MASTER}
                            />
                        </div>
                        <div className="col-4">
                            <ScrumMasterPanel />
                        </div>
                    </div>
                </div>
            </Layout>
        );
    }
    reRenderComponent = (): void => {
        this.props.getSessionsFromLocalStorage();
    };
}

const mapStateToProps = (state: IGlobalState) => ({
    activeUserStory: getActiveUserStory(state),
    selectedSession: getSelectedSession(state),
});

export const mapDispatchToProps = (dispatch: Function) => {
    return {
        setSelectedSession: (id: string) => {
            dispatch(setSelectedSession(id));
        },
        giveStoryPoint: (userType: UserTypeEnum, storyPoint?: number) => {
            dispatch(giveStoryPoint(userType, storyPoint));
        },
        getSessionsFromLocalStorage: () => {
            dispatch(getSessionsFromLocalStorage());
        },
    };
};

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps,
    )(SessionForScrumMaster),
);
