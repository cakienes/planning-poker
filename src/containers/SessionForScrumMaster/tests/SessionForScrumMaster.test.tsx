import React from 'react';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import { UserTypeEnum } from '../../../helper/Enum';
import {
    getSessionsFromLocalStorage,
    giveStoryPoint,
    setSelectedSession,
} from '../../../redux/session/session.actions';
import ISessionForScrumMasterProps from '../interface/ISessionForScrumMasterProps';
import { mapDispatchToProps, SessionForScrumMaster } from '../SessionForScrumMaster';

jest.mock('uuid', () => {
    return jest.fn().mockImplementation(() => {
        return 1;
    });
});

jest.mock('react-redux-toastr', () => {
    return {
        toastr: {
            error: jest.fn(),
        },
    };
});

jest.useFakeTimers();

describe('SessionForScrumMaster.tsx', () => {
    let renderer: ShallowRenderer;
    const getSessionsFromLocalStorageMock = jest.fn();
    const giveStoryPointMock = jest.fn();
    const setSelectedSessionMock = jest.fn();

    let sessionForScrumMasterProps: ISessionForScrumMasterProps = {} as any;

    beforeEach(() => {
        renderer = createRenderer();
        sessionForScrumMasterProps = {
            history: {} as any,
            location: {} as any,
            match: {
                params: {},
            } as any,
            staticContext: {} as any,
            getSessionsFromLocalStorage: getSessionsFromLocalStorageMock,
            giveStoryPoint: giveStoryPointMock,
            setSelectedSession: setSelectedSessionMock,
            activeUserStory: undefined,
            selectedSession: undefined,
        };
    });

    it('renders correctly SessionForScrumMaster', async () => {
        expect(renderer.render(<SessionForScrumMaster {...sessionForScrumMasterProps} />)).toMatchSnapshot();
    });

    it('renders correctly SessionForScrumMaster while selectedSession.optionalUrl is not defined', async () => {
        sessionForScrumMasterProps = {
            ...sessionForScrumMasterProps,
            selectedSession: {
                id: 'id',
                developers: [],
                numberOfVoters: 2,
                optionalUrl: undefined,
                sessionName: 'sessionName',
                userStories: [],
            },
        };
        expect(renderer.render(<SessionForScrumMaster {...sessionForScrumMasterProps} />)).toMatchSnapshot();
    });

    it('renders correctly SessionForScrumMaster while selectedSession.optionalUrl is defined', async () => {
        sessionForScrumMasterProps = {
            ...sessionForScrumMasterProps,
            selectedSession: {
                id: 'id',
                developers: [],
                numberOfVoters: 2,
                optionalUrl: 'optionalUrl',
                sessionName: 'sessionName',
                userStories: [],
            },
        };
        expect(renderer.render(<SessionForScrumMaster {...sessionForScrumMasterProps} />)).toMatchSnapshot();
    });

    it('componentDidMount of SessionForScrumMaster', async () => {
        renderer.render(<SessionForScrumMaster {...sessionForScrumMasterProps} />);
        const instance: any = renderer.getMountedInstance();

        instance.reRenderComponent = jest.fn();

        instance.componentDidMount();

        expect(setInterval).toHaveBeenCalledWith(instance.reRenderComponent, 1000);
    });

    it('componentDidMount of SessionForScrumMaster while id is defined', async () => {
        sessionForScrumMasterProps = {
            ...sessionForScrumMasterProps,
            match: {
                params: {
                    id: 'id',
                },
            } as any,
        };

        renderer.render(<SessionForScrumMaster {...sessionForScrumMasterProps} />);
        const instance: any = renderer.getMountedInstance();

        instance.componentDidMount();

        expect(setSelectedSessionMock).toHaveBeenCalledWith('id');
    });

    it('reRenderComponent of SessionForScrumMaster', async () => {
        renderer.render(<SessionForScrumMaster {...sessionForScrumMasterProps} />);
        const instance: any = renderer.getMountedInstance();

        instance.reRenderComponent();

        expect(getSessionsFromLocalStorageMock).toHaveBeenCalled();
    });

    describe('mapDispatchToProps', () => {
        it('should dispatch getSessionsFromLocalStorage when called', () => {
            const dispatchMock = jest.fn();
            const result = mapDispatchToProps(dispatchMock);
            result.getSessionsFromLocalStorage();
            expect(JSON.stringify(dispatchMock.mock.calls[0][0])).toEqual(
                JSON.stringify(getSessionsFromLocalStorage()),
            );
        });

        it('should dispatch setSelectedSession when called', () => {
            const dispatchMock = jest.fn();
            const result = mapDispatchToProps(dispatchMock);
            result.giveStoryPoint(UserTypeEnum.SCRUM_MASTER, 5);
            expect(JSON.stringify(dispatchMock.mock.calls[0][0])).toEqual(
                JSON.stringify(giveStoryPoint(UserTypeEnum.SCRUM_MASTER, 5)),
            );
        });

        it('should dispatch setSelectedSession when called', () => {
            const dispatchMock = jest.fn();
            const result = mapDispatchToProps(dispatchMock);
            result.setSelectedSession('id');
            expect(JSON.stringify(dispatchMock.mock.calls[0][0])).toEqual(JSON.stringify(setSelectedSession('id')));
        });
    });
});
