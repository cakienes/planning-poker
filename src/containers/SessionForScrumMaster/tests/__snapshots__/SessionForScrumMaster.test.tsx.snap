// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionForScrumMaster.tsx renders correctly SessionForScrumMaster 1`] = `
<Layout>
  <div
    className="session"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="linkHolder"
        >
          <span>
            Link for developers: 
          </span>
          <span
            className="font-weight-bold"
          >
            http://localhost/session/undefined/developer
          </span>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <UserStoryList />
      </div>
      <div
        className="col-4"
      >
        <ActiveStory
          giveStoryPoint={[MockFunction]}
          type="SCRUM_MASTER"
        />
      </div>
      <div
        className="col-4"
      >
        <Memo(ConnectFunction) />
      </div>
    </div>
  </div>
</Layout>
`;

exports[`SessionForScrumMaster.tsx renders correctly SessionForScrumMaster while selectedSession.optionalUrl is defined 1`] = `
<Layout>
  <div
    className="session"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="linkHolder"
        >
          <span>
            Link for developers: 
          </span>
          <span
            className="font-weight-bold"
          >
            http://localhost/optionalUrl/developer
          </span>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <UserStoryList
          selectedSession={
            Object {
              "developers": Array [],
              "id": "id",
              "numberOfVoters": 2,
              "optionalUrl": "optionalUrl",
              "sessionName": "sessionName",
              "userStories": Array [],
            }
          }
        />
      </div>
      <div
        className="col-4"
      >
        <ActiveStory
          giveStoryPoint={[MockFunction]}
          selectedSession={
            Object {
              "developers": Array [],
              "id": "id",
              "numberOfVoters": 2,
              "optionalUrl": "optionalUrl",
              "sessionName": "sessionName",
              "userStories": Array [],
            }
          }
          type="SCRUM_MASTER"
        />
      </div>
      <div
        className="col-4"
      >
        <Memo(ConnectFunction) />
      </div>
    </div>
  </div>
</Layout>
`;

exports[`SessionForScrumMaster.tsx renders correctly SessionForScrumMaster while selectedSession.optionalUrl is not defined 1`] = `
<Layout>
  <div
    className="session"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="linkHolder"
        >
          <span>
            Link for developers: 
          </span>
          <span
            className="font-weight-bold"
          >
            http://localhost/session/id/developer
          </span>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <UserStoryList
          selectedSession={
            Object {
              "developers": Array [],
              "id": "id",
              "numberOfVoters": 2,
              "optionalUrl": undefined,
              "sessionName": "sessionName",
              "userStories": Array [],
            }
          }
        />
      </div>
      <div
        className="col-4"
      >
        <ActiveStory
          giveStoryPoint={[MockFunction]}
          selectedSession={
            Object {
              "developers": Array [],
              "id": "id",
              "numberOfVoters": 2,
              "optionalUrl": undefined,
              "sessionName": "sessionName",
              "userStories": Array [],
            }
          }
          type="SCRUM_MASTER"
        />
      </div>
      <div
        className="col-4"
      >
        <Memo(ConnectFunction) />
      </div>
    </div>
  </div>
</Layout>
`;
